### ISSUE TITLE
Error during search: list index out of range

### ISSUE THREAD

**Issue by dondiegorivera** [TheBlewish/Automated-AI-Web-Researcher-Ollama#44]
Hi, 

I'm getting an error during search: "Error during search: list index out of range" since I pulled the latest version. 

The original worked fine. I am using WSL Ubuntu + phi3.5:3.8b-mini-instruct-q8_0. The result file does not have any search result.  According to the thread on reddit, other users are experiencing the same issue. 

Any hints what to look for? 

Thanks a lot.

**Comment by dondiegorivera** [comment]
My bad, API keys were not configured. Please close the issue.

**Comment by palencar** [comment]
Hi!

I had this same issue with Ubuntu and llama3.1:8b.

I solved it by exporting the variables on the .env file before running.
In my case:
export TAVILY_API_KEY="tvly-..."

**Comment by TOoSmOotH** [comment]
Same here. Tried exporting the .env but no luck

**Comment by synth-mania** [comment]
I'm pretty sure merging PR #1 is what broke this

**Comment by jziggles** [comment]
I am using linux mint with phi3:14b-medium-128k-instruct-q5_K_M.  I was not able to fix it by adding export to the .env file but then adding the export to my .bashrc did fix it.  Just make sure to confirm with echo $TAVILY_API_KEY I think that will work.

**Comment by jziggles** [comment]
> Interesting. It's disappointing that the repo owner and the author of #1 haven't even been able to comment on these issues. Im already working on my own hard fork (synth-mania/LLM-researcher-old) which won't be ready for awhile because I'm rewriting just about everything, but maybe I'll make a fork of the project as it currently stands to try to fix some of these issu

Sounds exciting!  Incorporating a pubmed api component would drastically increase the utility of this product I believe, though the existing search component does seem to hit ncbi anyways (I just started playing with it).  I don't know enough yet about how it works but the idea is really great.  But automated search and synthesis of update pubmed and, broadly, field/respected journals based on a prompt is pretty awesome if it works well...

**Comment by dondiegorivera** [comment]
Thanks all. @jziggles 's solution worked for me.

**Comment by TheBlewish** [comment]
Hey everyone! 

Whoops, my bad! 😅 I totally messed up and accidentally merged some big changes that added a bunch of API stuff. This wasn't supposed to happen it wasn't in the description of the changes I should've been more careful!

I've rolled it back to how it was before so it should work like it used to - just Ollama and openAI end points, basic web scraping, no fancy API stuff needed.

If it's still acting weird, just reopen this issue or make a new one and let me know what's up. 

Sorry about that! goodness me!

Also for those that still want to access that merged version it is still available in a second branch but the main has now been reverted!
